### ISSUE TITLE
[Kubuntu 23.10] Godot doesn't register the tap-to-click left-click (normal button-click works) and it doesn't close right-click menus as the result. They're then also drawn on top of other apps.

### ISSUE THREAD

**Issue by Bobbium** [godotengine/godot#88118]
### Tested versions

- Reproducible in every version after 4.0

### System information

Kubuntu 23.10, Godot 4.*, Forward+, AMD Ryzen 5500U

### Issue description

This only started happening when I switched to Kubuntu 23.10, it had never appeared before then.

It seems like the kernel versions that 23.10 introduced caused some bugs, because I also had a similar bug to this one #88113, where I explained my issue in detail, though the most recent kernel update fixed it. Also, as opposed to that one, this one happens both on X11 and Wayland.

### Steps to reproduce

Open up any right-click menu.

### Minimal reproduction project (MRP)

N/A

**Comment by akien-mga** [comment]
Just to be sure, can you confirm that you do have Tap-to-click enabled in the settings?
![image](https://github.com/godotengine/godot/assets/4701338/799a9845-d43b-474e-acb3-95e6bd0d930f)

**Comment by Bobbium** [comment]
Yup, it's enabled and works fine everywhere else.

**Comment by Bobbium** [comment]
Well, I've just found out that it doesn't happen when Tap-and-drag is enabled. Never thought of that possibility, I've always had it disabled.

**Comment by akien-mga** [comment]
I tested and I can reproduce the issue on Fedora 39 if I disable Tap-and-drag.

I don't think your kernel update is the problem, but you likely got a KDE update that added the Tap-and-drag option and changed the default Tap-to-click behavior. I also don't remember seeing this option last time I configured Tap-to-click a year ago or so (though I might misremember).

I'm not sure what Tap-and-drag is meant to do, but apparently it doesn't send the right events for Godot to register when there's a click outside a popup dialog.

**Comment by Bobbium** [comment]
Yeah, it's probably a KDE bug. I believe the option was there in the last version, 23.04, though, and it worked fine back then. Maybe some back-end stuff that concern it were changed though (or I'm misremembering). Also yeah, it doesn't sound like it should affect Godot really, weird.

**Comment by TheV360** [comment]
Maybe bumping a very old issue, but I have the same issue. However, despite the title, this happens to me on GNOME as well. I open a popup dialog, and I can't close it -- meanwhile, the actual editor is still handling my taps.

I have a video that may not illustrate this perfectly, but still does a good job. I only tap for the first 20 seconds or so, and you see that the popup menu I have opened with a tap does not close with a tap. In the last 10 or so seconds, I switch to clicking with the physical button above my touchpad for a bit, and that eats the click event and closes the popup menu without sending the event to the editor. (Note that all the while, my selection doesn't change!) (In the last second, I start using touchpad taps again.)

https://github.com/user-attachments/assets/5c3405b5-d078-40d9-af56-14ba369e543b

openSUSE Tumbleweed, Godot 4.4, Compatibility renderer, AMD Ryzen 5 5625U with Radeon Graphics × 12
